src/state/slices/opportunitiesSlice/opportunitiesSlice.ts

```diff
@@ -57,30 +57,27 @@ export const opportunities = createSlice({
           ]
         })
       },
-      // Use the `prepareAutoBatched` utility to automatically
-      // add the `action.meta[SHOULD_AUTOBATCH]` field the enhancer needs
-      prepare: prepareAutoBatched<GetOpportunityUserDataOutput>(),
-    },
-    upsertUserStakingOpportunities: {
-      reducer: (draftState, { payload }: { payload: GetOpportunityUserStakingDataOutput }) => {
+    ),
+    upsertUserStakingOpportunities: create.preparedReducer(
+      prepareAutoBatched<GetOpportunityUserStakingDataOutput>(),
+      (draftState, { payload }: { payload: GetOpportunityUserStakingDataOutput }) => {
         draftState.userStaking.byId = Object.assign(draftState.userStaking.byId, payload.byId)
         draftState.userStaking.ids = Object.keys(draftState.userStaking.byId) as UserStakingId[]
       },
-
-      // Use the `prepareAutoBatched` utility to automatically
-      // add the `action.meta[SHOULD_AUTOBATCH]` field the enhancer needs
-      prepare: prepareAutoBatched<GetOpportunityUserStakingDataOutput>(),
-    },
-    invalidateUserStakingOpportunity: {
-      reducer: (draftState, { payload }: { payload: UserStakingId }) => {
+    ),
+    invalidateUserStakingOpportunity: create.preparedReducer(
+      prepareAutoBatched<UserStakingId>(),
+      (draftState, { payload }: { payload: UserStakingId }) => {
         const userStakingId = payload
         delete draftState.userStaking.byId[userStakingId]
       },
-
-      // Use the `prepareAutoBatched` utility to automatically
-      // add the `action.meta[SHOULD_AUTOBATCH]` field the enhancer needs
-      prepare: prepareAutoBatched<UserStakingId>(),
-    },
+    ),
+  }),
+  selectors: {
+    selectLpOpportunitiesById: state => state.lp.byId,
+    selectStakingOpportunitiesById: state => state.staking.byId,
+    selectUserStakingIds: state => state.userStaking.ids,
+    selectStakingByAccountId: state => state.staking.byAccountId,
```

Comment:
- can add `selectStakingIds` and replace `stakingSelectors:L52`
- can add `selectUserStakingOpportunitiesById` and replace `stakingSelectors:L77`

Reply:
a30c9b8d2ca4c2879396c2d9af051f228d774f82